test/integration/vaultdynamicsecret_integration_test.go

```diff
@@ -363,10 +357,19 @@ func TestVaultDynamicSecret(t *testing.T) {
 		}
 		keys := resp.Data["keys"].([]interface{})
 		if len(keys) > 0 {
-			return "", fmt.Errorf("Leases still found: %d", len(keys))
+			// Print out the lease ids that are still found to make debugging easier.
+			return "", fmt.Errorf("leases still found: %v", keys)
 		}
 		return "", nil
 	})
+	// Delete custom resources which were created before tf destroy removes the k8s namespace.
+	// This includes the VaultAuthMethods/Connections which are required for revocation to complete,
+	// so do it after we test for revocation.
+	for _, c := range created {
```

Comment:
So the revocation is being triggered a reason other than the CR being deleted?

Reply:
It turns out that only the auth+conn objects end up in the `created` list, and we have a secondary list called `objsCreated` which holds the VDS objects which gets cleaned up as well in a `t.Cleanup()` function inside of each test: https://github.com/hashicorp/vault-secrets-operator/blob/a3a2d18a03caebd4de4688e532a0c3fb2b3884fd/test/integration/vaultdynamicsecret_integration_test.go#L224.

My understanding is that the `t.Cleanup()` [inside](https://github.com/hashicorp/vault-secrets-operator/blob/a3a2d18a03caebd4de4688e532a0c3fb2b3884fd/test/integration/vaultdynamicsecret_integration_test.go#L224) of the `t.Run()` will trigger at the end of `t.Run()`, and the `t.Cleanup()` [which holds](https://github.com/hashicorp/vault-secrets-operator/blob/a3a2d18a03caebd4de4688e532a0c3fb2b3884fd/test/integration/vaultdynamicsecret_integration_test.go#L98) the `tf destroy` will trigger at the end of the test??

Reply:
Moved this up into the main `t.Cleanup()`  in 179e8ef